server/lib/index.js

```diff
@@ -4,11 +4,13 @@ import { of, fromPromise, Rejected, Resolved } from 'hyper-async'
 import { dispatch } from '../lib/dispatch.js'
 import { writeInteraction } from '../lib/write-interaction.js'
 import { isVouched } from '../lib/is-vouched.js'
+import { sendMessage } from './send-message.js'
 
 export function vouch(startdate, address) {
   const sixMonthsAgo = sub(new Date(), { months: 6 })
   if (isBefore(parseISO(startdate), sixMonthsAgo)) {
     return of({ address })
+      .chain(fromPromise(sendMessage))
```

Comment:
Why is this above the already vouched check?

Wouldn't that create problems for people vouching that have already been vouched? I would think it would be a better fit to go after the `write-interaction` chain?

What do you think?

Reply:
Vouching with Vouch DAO is idempotent, so having it first isn't a problem.

But there is a guard/check `isVouched` on the next line to stop the `dispatch` function from writing multiple vouches to Arweave/Warp.

If I put `sendMessage` after `writeInteraction` then anyone who is already vouched on Arweave/Warp will not be able to use the Vouch X UI to re-vouch and add themselves to Vouch DAO on AO because `isVouched` will reject.

That was my thinking.

Reply:
Moved to the end of the chain now.